src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -781,7 +781,8 @@ await FixReferencesAsync(document, changeNamespaceService, addImportService, ref
 
                 var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
                 var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-                root = addImportService.AddImports(compilation, root, contextLocation, imports, placeSystemNamespaceFirst, cancellationToken);
+                var generator = document.GetLanguageService<SyntaxGenerator>();
+                root = addImportService.AddImports(compilation, root, contextLocation, imports, generator, placeSystemNamespaceFirst, cancellationToken);
```

Comment:
i still find this confusing.  how is it that the service can use a generator, but cannot *retrieve* the generator itself?  They're just singletons.  so the VB service could just retrieve the VBGenerator and the C# woudl can do the same.  I really don't understand the need to pass these in, and i would prefer this change not happen unless absolutely necessary.

Reply:
If we move VB and C# syntax generator implementations to the shared layer, we can revert this code. Otherwise, we need to retain it as they are internal types defined in Workspaces layer.